pkg/skaffold/kubernetes/portforward/port_forward_entry.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package portforward
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+type portForwardEntry struct {
+	resourceVersion        int
+	resource               latest.PortForwardResource
+	podName                string
+	containerName          string
+	portName               string
+	localPort              int32
+	automaticPodForwarding bool
+
+	cancel context.CancelFunc
+}
+
+// key is an identifier for the lock on a port during the skaffold dev cycle.
+// if automaticPodForwarding is set, we return a key that doesn't include podName, since we want the key
+// to be the same whenever pods restart
+func (p *portForwardEntry) key() string {
+	if p.automaticPodForwarding {
```

Comment:
this smells a little bit like that we should probably have two separate types for Resource and Pod entries instead of an if statement?

Reply:
Yah I thought about that, but it seemed like a lot of extra code just for two different types of keys. This way simplified both [forwarding](https://github.com/GoogleContainerTools/skaffold/pull/2215/files#diff-93105af8f6782cbf38d2aa66bb239006R59) the `portForwardEntry` and the unit tests.